server/src/clientglue.js

```diff
@@ -26,6 +26,32 @@ exports.setModel = function (data) {
       isExtInstalled = false;
       render();
     }, 2000);
+    let toolbar = document.getElementById("toolbar");
+    let navigateToolbar = document.getElementById("navigate-toolbar");
+    let fullPageButton = document.getElementById("full-page-button");
+    let fullPageButtonScrollable = document.getElementById("full-page-button-scrollable");
+    let frameElement = document.getElementById("frame");
+    window.onscroll = function (e) {
+      if (e.pageY > 0) {
+        toolbar.style.visibility = "visible";
+        navigateToolbar.style.visibility = "visible";
+        let frameOffset = frameElement.getBoundingClientRect().top + window.scrollY;
+        let toolbarHeight = toolbar.clientHeight;
+        let visibleHeight = window.innerHeight - toolbarHeight;
+        let frameTop = frameOffset - toolbarHeight;
+        if (e.pageY >= frameTop - visibleHeight - 16) {
```

Comment:
It's actually 1/2 the height of the element. I changed the code to calculate it.